src/api/operations/externalFormSubmission.fragments.graphql

```diff
@@ -0,0 +1,13 @@
+fragment ExternalFormSubmissionFields on ExternalFormSubmission {
+  id
+  submittedAt
+  spam
+  status
+  notes
+  definition {
+    ...FormDefinitionFields
```

Comment:
This gave me pause, because I don't think there's anywhere else where we load the full FormDefinition JSON on each row in a table. This is going to return a lot of data that is repeated, because we're only ever showing a table for one form definition.

This is really just an optimization, but I think it's worth doing. The way I would go about it is:
* In the `ProjectExternalFormSubmissions` component, fetch the FormDefinition when an identifier is selected. I think this would require adding a new query `external_form_definition(identifier:)`  since we currently only do lookups by role or id.
* Stop resolving definition on each row item
* Replace calls to `selected.definition.definition` with the single definition that was passed to `ProjectExternalSubmissionsTable` as a prop

Reply:
@gigxz  sounds good, I created a PR here for the backend change to add that query: https://github.com/greenriver/hmis-warehouse/pull/4158/files

The frontend is still giving me some trouble ("Uncaught Invariant Violation: Missing field 'cacheKey' while extracting keyFields from {"identifier":"tchc_prevention_screening","__typename":"FormDefinition"}") altho the `cacheKey` does appear to exist ... I'll keep poking at it!

Reply:
Hi @martha - we configure apollo to use definition.cacheKey for FormDefinitions (see the typePolicy in src/providers/apolloClient.tsx:150). Maybe that the cacheKey isn't being selected in your query? Happy to pair if it would be helpful

Reply:
thanks @ttoomey for the help!! :)